webui/src/lib/api/index.js

```diff
@@ -641,6 +641,18 @@ class Objects {
         return response.text()
     }
 
+    async getPresignedUrl(repoId, ref, path) {
```

Comment:
Consider levreging the previous function instead of calling the same endpoint from another location.

Also there are a few requests that involves presigned urls, like upload, list, etc.
So I think it's better to name it explicitly (the `get` at the start can mean a lot of things). I couldn't find something nice:
```suggestion
    async getPresignedUrlForDownload(repoId, ref, path) {
```

Reply:
Done and done